src/compiler/program.ts

```diff
@@ -430,13 +430,14 @@ namespace ts {
         return program;
 
         function getCommonSourceDirectory() {
-            if (typeof commonSourceDirectory === "undefined") {
-                if (options.rootDir && checkSourceFilesBelongToPath(files, options.rootDir)) {
+            if (commonSourceDirectory === undefined) {

```

Comment:
I think this is the section @sheetalkamat is referring to: https://github.com/Microsoft/TypeScript/blob/4a906143c62fcbfe113bf3f1cc607c488df47400/src/compiler/sourcemap.ts#L171-L180